redisvl/redis/utils.py

```diff
@@ -30,14 +33,22 @@ def convert_bytes(data: Any) -> Any:
     return data
 
 
-def array_to_buffer(array: List[float], dtype: Any = np.float32) -> bytes:
+def array_to_buffer(array: List[float], dtype: str) -> bytes:
     """Convert a list of floats into a numpy byte string."""
-    return np.array(array).astype(dtype).tobytes()
+    if dtype.upper() not in {v.value for v in VectorDataType}:
```

Comment:
enums can be used to check membership --> `VectorDataType(dtype.upper())`

this will throw a `ValueError` if the value is not a member of the enum.

Reply:
This makes membership check cleaner. I still want to raise our own value error so we can list out the accepted dtypes as the default ValueError doesn't do that.

Reply:
Yep -- like this?
```python
try:
    VectorDataType(dtype.upper())
except:
    raise ValueError("custom error msg")
```